packages/wiggle/src/WiggleTrack/model.ts

```diff
@@ -189,7 +188,9 @@ const stateModelFactory = (configSchema: any) =>
                   const aborter = new AbortController()
                   self.setLoading(aborter)
                   const stats = await getStats(aborter.signal)
-                  if (isAlive(self)) {
+
+                  // check if stats returned from empty region
+                  if (isAlive(self) && !stats.emptyRegion) {
```

Comment:
I feel concerned about the stats code having to be concerned about empty region. I feel like maybe a way to approach it is by making it something relevant to the track's logic. Example https://github.com/GMOD/jbrowse-components/compare/792_multiple_renderers_per_track...792_multiple_renderers_per_track_suggestion

I know that is picky but it seems like the stats code should be more or less just about the data whereas this empty region weirdness is really more about the track being setup

Reply:
Valid concern and I agree. Thanks for the suggestion code, have implemented and tested around. Addresses the issue of premature empty updating of stats without having to check for a specific stats object returned